packages/core/src/fhirpath/atoms.ts

```diff
@@ -55,13 +55,10 @@ export class SymbolAtom implements Atom {
     if (this.name === '$this') {
       return context;
     }
-    return context
-      .map((e) => this.#evalValue(e))
-      .flat()
-      .filter((e) => e?.value !== undefined) as TypedValue[];
+    return context.flatMap((e) => this.evalValue(e)).filter((e) => e?.value !== undefined) as TypedValue[];
   }
 
-  #evalValue(typedValue: TypedValue): TypedValue[] | TypedValue | undefined {
+  private evalValue(typedValue: TypedValue): TypedValue[] | TypedValue | undefined {
```

Comment:
Another interesting revelation from the Node.js profiler is that using the built-in JavaScript `#` for private has significantly worse performance than using TypeScript's `private`. 

TypeScript's `private` only works at compile time.  There is no runtime cost at all.

JavaScript's `#` is transpiled to a polyfill using `WeakMap` and a bunch of other complexity.  I'm tempted to say we should remove all `#` private methods and go back to TypeScript `private`.